File: c++/0459.cpp
// Link: https://leetcode.com/problems/repeated-substring-pattern/description/
// Runtime: 6 ms
// Memory: 14.21 MB

class Solution {
public:
  bool repeatedSubstringPattern(const std::string &s) {
    std::string doubleString = s + s;

    return doubleString.substr(1, static_cast<int>(doubleString.size()) - 2)
               .find(s) != -1;
  }
};

File: python/0459.py
# Link: https://leetcode.com/problems/repeated-substring-pattern/description/
# Runtime: 41 ms
# Memory: 16.48 MB


class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        double_string = s + s
        if s in double_string[1:-1]:
            return True
        else:
            return False
